src/quicklist.c

```diff
@@ -905,6 +909,9 @@ REDIS_STATIC quicklistNode *_quicklistSplitNode(quicklistNode *node, int offset,
     /* Copy original listpack so we can split it */
     memcpy(new_node->entry, node->entry, zl_sz);
 
+    /* Need positive offset for calculating extent below. */
+    if (offset < 0) offset = node->count + offset;
```

Comment:
can we specify in the top comment in which conditions this bug will trigger?
i.e. so that users can know if they're affected and must upgrade.
will it always result in a crash or also sometimes in messing up the data?

Reply:
i meant something that's more useful to end users.
like when a certain argument for a certain command is negative.
some way that users can use to figure out if they're using it this way and if they're at risk.

Reply:
ohh, i see you did mention it as the end.
so only LSET with negative index and a big (> 1gb) object?

Reply:
Yes, LSET with a bit object is the only way that will trigger this crash.